resources/lib/modules/updates.py

```diff
@@ -483,21 +481,122 @@ def check_updates_v2(self, force=False):
             version = oe.VERSION
         url = f'{self.UPDATE_REQUEST_URL}?i={oe.url_quote(systemid)}&d={oe.url_quote(oe.DISTRIBUTION)}&pa={oe.url_quote(oe.ARCHITECTURE)}&v={oe.url_quote(version)}&f={oe.url_quote(self.hardware_flags)}'
         if oe.BUILDER_NAME:
-           url += f'&b={oe.url_quote(oe.BUILDER_NAME)}'
+            url += f'&b={oe.url_quote(oe.BUILDER_NAME)}'
 
         log.log(f'URL: {url}', log.DEBUG)
         update_json = oe.load_url(url)
         log.log(f'RESULT: {repr(update_json)}', log.DEBUG)
-        if update_json:
-            update_json = json.loads(update_json)
-            self.last_update_check = time.time()
-            if 'update' in update_json['data'] and 'folder' in update_json['data']:
-                self.update_file = self.UPDATE_DOWNLOAD_URL % (update_json['data']['folder'], update_json['data']['update'])
-                if self.struct['update']['settings']['UpdateNotify']['value'] == '1':
-                    oe.notify(oe._(32363), oe._(32364))
-                if self.struct['update']['settings']['AutoUpdate']['value'] == 'auto' and force == False:
-                    self.update_in_progress = True
-                    self.do_autoupdate(None, True)
+
+        # statistics sent to UPDATE_REQUEST_URL, but discard the response.
+        update_json = None
+
+        # get releases.json
+        release_data = self.get_json()
+        self.last_update_check = time.time()
+
+        # releases.json is empty
+        if release_data is None:
+            log.log("releases.json is empty. Failed to retrieve?", log.WARNING)
+            return
+
+        # devel versions manage their own updates
+        if oe.VERSION.startswith("devel"):
+            log.log("Update check skipped because this is a development build.", log.INFO)
+            return
+
+        # parse installed VERSION for comparison
+        version_major = int(oe.VERSION.split(".")[0])
+        version_minor = int(oe.VERSION.split(".")[1])
+        version_bugfix = int(oe.VERSION.split(".")[2])
+
+        # check highest bugfix release of installed branch
+        highest_device_release = 0
+
+        for device_release in release_data[f"{oe.DISTRIBUTION}-{oe.VERSION_ID}"]['project'][oe.ARCHITECTURE]['releases'].keys():
+            device_release = int(device_release)
+            if device_release > highest_device_release:
+                highest_device_release = device_release
+
+        highest_device_release = str(highest_device_release)
+
+        bugfix_filename = release_data[f"{oe.DISTRIBUTION}-{oe.VERSION_ID}"]['project'][oe.ARCHITECTURE]['releases'][highest_device_release]['file']['name']
+        # assumes filename format is "distribution-device.arch-version.tar"
+        bugfix_filename_version = bugfix_filename.split("-")[2].rstrip(".tar")
```

Comment:
rstrip() is stripping all characters of the string (yes, strip() is misused in the current update code too). We may need to add something like [rchop()](https://stackoverflow.com/questions/3663450/remove-substring-only-at-the-end-of-string).